src/doc/performance-tuning.md

```diff
@@ -1,91 +1,185 @@
+   0:         41          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0  IR-IO-APIC    2-edge      timer
+   8:          1          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0          0  IR-IO-APIC    8-edge      rtc0
+...
 ```
 
-The allocated memory must match the memory-backend-file size (example shows 8GB). While qemu will fail to boot if there isn't enough hugepages allocated, it is recommended to have some spare and note that the pages are split amongst the NUMA nodes. Check the paragraph on NUMA in this  document.
-It is recommended to specify the qemu option '-realtime mlock=on', despite it being the default. This ensures memory doesn't get swapped out. 
-
-## Hardware / BIOS
-### Disable Hyper-Threading
-Disable hyper-threading (HT) in the BIOS. Even with isolating the correct hyper-threaded CPU's, can create latency spikes, leading to packet loss, when enabled. (TODO: do we have one of the automated tests showing this?)
-According to [Intel on Hyper-Threading](http://www.intel.com/content/www/us/en/architecture-and-technology/hyper-threading/hyper-threading-technology.html): "Intel® Hyper-Threading Technology (Intel® HT Technology) uses processor resources more efficiently, enabling multiple threads to run on each core. As a performance feature, it also increases processor throughput, improving overall performance on threaded software.". Snabb runs single threaded, so can't benefit directly from HT. 
-### Performance Profile set to Max 
-Servers are optimized for energy efficiency. While this is great for application servers, Virtual Network Functions like Snabb benefit from performance optimized settings. Each vendor offers different BIOS settings to enable or disable energy efficiency settings or profiles. They are typically named "Max performance", "Energy Efficiency" and "Custom".  Select "Max performance" for latency sensitive Snabb use. 
-### Turbo Mode
-Intel Turbo Boost Technology allows processor cores to run faster than the rated operating frequency if they're operating below power, current, and temperature specification limits. (TODO: impact not yet analyzed on Snabb, nor if it is controlled by the performance profile). 
-## Linux Kernel
 ### Disable IOMMU
-Sandybridge CPUs have a known issue on its IOTBL huge page support, impacting small packet performance.  Newer CPUs don't have this issue.
-(TODO, only found this info here: [http://dpdk.org/ml/archives/dev/2014-October/007411.html]()
-(TODO: pass through mode: [https://lwn.net/Articles/329174/]())
 
-Add IOMMU=pt (pass through) to the kernel:
+The oldest CPU supported by Snabb is the Sandy Bridge series, and this
```

Comment:
I think it's worth documenting that on older supported processors, having IOMMU on can (IIRC) lead to 30%-100% packet loss - that's pretty dramatic, and was hard-won non-obvious knowledge that it would be sad to have people have to troubleshoot from scratch again.